crowbar_framework/app/models/api/storage.rb

```diff
@@ -15,6 +15,44 @@
 #
 
 module Api
-  class Storages
+  class Storage < Tableless
+    def repocheck
+      {}.tap do |ret|
+        ret["ceph"] = {
+          "available" => true
+        }
+
+        platform = Api::Upgrade.new.target_platform(platform_exception: :ceph)
+
+        ["os", "ceph"].each do |feature|
+          ceph_node_architectures.each do |architecture|
+            unless ::Crowbar::Repository.provided_and_enabled?(feature,
+                                                               platform,
+                                                               architecture)
+              ::Openstack::Upgrade.enable_repos_for_feature(feature, Rails.logger)
+            end
+            available, repolist = ::Crowbar::Repository.provided_and_enabled_with_repolist(
+              feature, platform, architecture
+            )
+            ret["ceph"]["available"] &&= available
+            ret["ceph"]["repos"] ||= {}
+            ret["ceph"]["repos"].deep_merge!(repolist.deep_stringify_keys)
+          end
+        end
+      end
+    end
+
+    protected
+
+    def ceph_node_architectures
+      ceph_proposal = Proposal.find_by(barclamp: "ceph", name: "default")
```

Comment:
I was careful to say "applied proposal" (I hope ;-)). You're looking at proposals, which are not necessarily applied.

Also, you should not restrict on the name, just in case we change the code to support multiple ceph proposals again (or in case someone named the proposal differently than "default").